### DISCUSSION TITLE
install the specified magento2 version

### DISCUSSION THREAD

**Discussion by bulemars** [magenx/Magento-2-server-installation#109] [General]
This script is great! thank you very much

Can I install the specified magento2 version? For example, magento2.4.3p3, please help me and give me a method

**Comment by artmouse** [comment]
> This script is great! thank you very much
> 
> Can I install the specified magento2 version? For example, magento2.4.3p3, please help me and give me a method

https://experienceleague.adobe.com/docs/commerce-operations/installation-guide/system-requirements.html

MagenX install script use PHP 8.1 but for Magento 2.4.3-p3 need PHP 7.4 and composer v1

**Comment by bulemars** [comment]
Can you help me modify this script so that he can download 2.4.3p3? Thank you.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "magenx/Magento-2-server-installation"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年11月6日(星期天) 下午5:07
***@***.***&gt;;
***@***.******@***.******@***.***&gt;;
主题:&nbsp;Re: [magenx/Magento-2-server-installation] help me (Discussion #109)





  
This script is great! thank you very much
 
Can I install the specified magento2 version? For example, magento2.4.3p3, please help me and give me a method
  
https://experienceleague.adobe.com/docs/commerce-operations/installation-guide/system-requirements.html
 
MagenX install script use PHP 8.1 but for Magento 2.4.3-p3 need PHP 7.4 and composer v1

**Comment by magenx** [comment]
yes we can add magento version selection

**Comment by bulemars** [comment]
I modify this part  
```
# Magento
MAGENTO_VERSION="2"
MAGENTO_VERSION_FULL=$(curl -s https://api.github.com/repos/magento/magento${MAGENTO_VERSION}/tags 2>&1 | head -3 | grep -oP '(?<=")\d.*(?=")')
MAGENTO_PROJECT="composer create-project --repository-url=https://repo.magento.com/ magento/project-community-edition=2.4.3-p3"
```

**Comment by magenx** [comment]
now you can select any magento version
also you need to make sure to change php version

**Comment by bulemars** [comment]
I found that you have added a version selection. Can you add the option of composer

**Comment by magenx** [comment]
`composer self-update --1`

**Reply by bulemars** [reply]
At which step is this command executed
